aip/0231.md

```diff
@@ -0,0 +1,86 @@
+---
+aip:
+  id: 231
+  state: reviewing
+  created: 2019-06-18
+permalink: /231
+redirect_from:
+  - /0231
+---
+
+# Batch Get
+
+Some APIs need to allow users to get a specific set of resources at a consistent
+time point (e.g. using a read transaction). A batch get method provides this
+functionality.
+
+## Guidance
+
+APIs **may** support Batch Get using the following pattern:
+
+```proto
+rpc BatchGetBooks(BatchGetBooksRequest) returns (BatchGetBooksResponse) {
+  option (google.api.http) = {
+    get: "/v1/shelves/-/books:batchGet"
+  };
+}
+```
+
+- The RPC's name **must** begin with `BatchGet`. The remainder of the
+  RPC name **should** be the plural form of the resource being retrieved.
+- The request and response messages **must** match the RPC name, with `-Request`
```

Comment:
Where did the convention of adding a hyphen for `-Request` and `-Response` suffix come from? When I first read it, it sounds like the full message should be something like `BatchGetBooks-Request`, which isn't the case.

Reply:
Uh, I think that was just my writing style which became a convention. :-) I am not married to it, but we should change it in other AIPs if it is confusing.

Reply:
I'll leave that for a separate pass, and I'll be consistent with the existing ones for now.